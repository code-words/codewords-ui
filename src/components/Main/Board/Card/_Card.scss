
.front {

  &.clickable {
    cursor: pointer;
  }

  &.red {
    background: $intel-red;
    border: 1px solid $intel-red-accent;
  }
  
  &.blue {
    background: $intel-blue;
    border: 1px solid $intel-blue-accent;
  }

  &.bystander {
    background: $intel-base;
    border: 1px solid $intel-base-accent;
  }

  &.assassin {
    background: $intel-assassin;
    border: 1px solid $intel-assassin-accent;
  }
}

.word-bg {
  background: #FFFFFF;
  width: calc(100% - 3px);
  display: flex;
  align-items: center;
  justify-content: center;

  .word-txt {
    text-align: center;
    text-transform: uppercase;
    font-size: .9em;
    font-weight: 500;
    padding: 10px 0;
    color: rgb(108, 93, 74);
  }
}