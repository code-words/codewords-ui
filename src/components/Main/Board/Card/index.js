import React, { Component } from 'react';
class Card extends Component {
  handleClick = e => {
    if (this.props.isActive && !this.props.card.type) {
      this.props.sendGuess({ id: e.currentTarget.id });
    }
  };
  
	render() {
		const { card, isActive, isHover } = this.props;
		console.log(isHover)
    const { type, flipped, word, id } = card;
		const intelCls = type ? type : '';
		const clickCls = !type && isActive && !flipped ? 'clickable' : null;
		const img = type ? require(`../../../../images/${type}.jpg`) : null;

		const content = flipped ? (
      <article className="card" key={`${word}`} id={id}>
				<img className="card" src={img} alt={type} />
			</article>
		) : (
			<article
				className={`card front ${intelCls} ${clickCls} ${isHover}-hover`}
				key={`${word}`}
				id={id}
				name={word}
				onClick={this.handleClick}>
				<div className="word-bg" key={`${word}-bg`}>
					<p className="word-txt" key={`${word}-txt`}>
						{word}
					</p>
				</div>
			</article>
		);

		return content;
	}
}

export default Card;
