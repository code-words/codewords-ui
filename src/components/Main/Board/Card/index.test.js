import Card from './index';
import React from 'react';
import { shallow } from 'enzyme'
 require('../../../../images/bystander.jpg');



describe('Card', () => {
    let mockCard = {type: true}
    let mockFn = jest.fn()
    let wrapper = shallow(<Card card = {mockCard} sendGuess={mockFn}/>)
    it('should matchsnapshot', () => {
        wrapper = shallow (<Card card={{type: "bystander"}} sendGuess={mockFn} />)
        wrapper.setState({ flipped: false});
        expect(wrapper).toMatchSnapshot()
    });
    it('should matchsnapshot', () => {
        let wrapper = shallow(<Card card={mockCard} isActive={false}/>)
        wrapper.setState({ flipped: true });
        expect(wrapper).toMatchSnapshot()
    });
    it('handleClick should be called on click', () => {
        const spy = jest.spyOn(wrapper.instance(), 'handleClick')
        let mockEvent = { target: { value: "", name: "title" } }
        wrapper.find('article').simulate('click', mockEvent);
        expect(spy).toHaveBeenCalled()
    });
    it('should send an id when is Active is true and card.type is false', () => {
        let mockCard = {type: false}
         wrapper = shallow(<Card card={mockCard} sendGuess={mockFn} isActive={true}/>)
        expect(mockFn).toHaveBeenCalled()
    });
    it('should not send an id when is Active is false and card.type is true', () => {
        let mockCard = {type: true}
         wrapper = shallow(<Card card={mockCard} sendGuess={mockFn} isActive={false}/>)
        expect(mockFn).toHaveBeenCalled()
    });

});