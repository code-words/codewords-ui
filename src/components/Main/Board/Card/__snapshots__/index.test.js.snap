// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should matchsnapshot 1`] = `ShallowWrapper {}`;

exports[`Card should matchsnapshot 2`] = `ShallowWrapper {}`;
